### PR TITLE
SUS-1403 | Remove page id from StaleResultException message

### PR DESCRIPTION
**Pull request by rogatty**
https://wikia-inc.atlassian.net/browse/SUS-1403

We have a flood of duplicated error reporter tickets. Let's generalize the message and pass the page id in the context instead.

/cc @TK-999 @macbre

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 12 issues
* MAJOR 5 major
* MINOR 6 minor
* INFO 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [MediaWikiService.php](https://github.com/Wikia/app/blob/36104c68929cde317e8996c06777a868b8a9071b/extensions/wikia/Search/classes/MediaWikiService.php): File "MediaWikiService.php" has 1024 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)
1. MAJOR [MediaWikiService.php#L16](https://github.com/Wikia/app/blob/36104c68929cde317e8996c06777a868b8a9071b/extensions/wikia/Search/classes/MediaWikiService.php#L16): Class "MediaWikiService" has 77 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. MAJOR [MediaWikiService.php#L475](https://github.com/Wikia/app/blob/36104c68929cde317e8996c06777a868b8a9071b/extensions/wikia/Search/classes/MediaWikiService.php#L475): Remove this unused "$page" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. MAJOR [MediaWikiService.php#L719](https://github.com/Wikia/app/blob/36104c68929cde317e8996c06777a868b8a9071b/extensions/wikia/Search/classes/MediaWikiService.php#L719): Remove this unused "$file" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. MAJOR [WikiaException.php#L40](https://github.com/Wikia/app/blob/36104c68929cde317e8996c06777a868b8a9071b/includes/wikia/nirvana/WikiaException.php#L40): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. MINOR [MediaWikiService.php#L169](https://github.com/Wikia/app/blob/36104c68929cde317e8996c06777a868b8a9071b/extensions/wikia/Search/classes/MediaWikiService.php#L169): Define a constant instead of duplicating this literal "action" 6 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [MediaWikiService.php#L183](https://github.com/Wikia/app/blob/36104c68929cde317e8996c06777a868b8a9071b/extensions/wikia/Search/classes/MediaWikiService.php#L183): Define a constant instead of duplicating this literal "query" 9 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [MediaWikiService.php#L586](https://github.com/Wikia/app/blob/36104c68929cde317e8996c06777a868b8a9071b/extensions/wikia/Search/classes/MediaWikiService.php#L586): Define a constant instead of duplicating this literal "allmessages" 4 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [MediaWikiService.php#L679](https://github.com/Wikia/app/blob/36104c68929cde317e8996c06777a868b8a9071b/extensions/wikia/Search/classes/MediaWikiService.php#L679): Define a constant instead of duplicating this literal "width" 9 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [MediaWikiService.php#L680](https://github.com/Wikia/app/blob/36104c68929cde317e8996c06777a868b8a9071b/extensions/wikia/Search/classes/MediaWikiService.php#L680): Define a constant instead of duplicating this literal "height" 9 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

File: includes/wikia/nirvana/WikiaException.php [hunk 1]
```diff
@@ -53,22 +53,26 @@ function report() {
  * @link http://pl2.php.net/manual/en/class.exception.php
  */
 class WikiaException extends WikiaBaseException {
-	public function __construct($message = '', $code = 0, Exception $previous = null) {
+	public function __construct( $message = '', $code = 0, Exception $previous = null, $extraContext = [] ) {
```
Comment (macbre) [inline includes/wikia/nirvana/WikiaException.php]: Let's provide the type hint for `$extraContent` argument

**Comment by WikiaTech** [comment]
SonarQube analysis reported 12 issues
* MAJOR 5 major
* MINOR 6 minor
* INFO 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [MediaWikiService.php](https://github.com/Wikia/app/blob/9132cb444f814e4ad657be02fed15fc60764d4d6/extensions/wikia/Search/classes/MediaWikiService.php): File "MediaWikiService.php" has 1024 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)
1. MAJOR [MediaWikiService.php#L16](https://github.com/Wikia/app/blob/9132cb444f814e4ad657be02fed15fc60764d4d6/extensions/wikia/Search/classes/MediaWikiService.php#L16): Class "MediaWikiService" has 77 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. MAJOR [MediaWikiService.php#L475](https://github.com/Wikia/app/blob/9132cb444f814e4ad657be02fed15fc60764d4d6/extensions/wikia/Search/classes/MediaWikiService.php#L475): Remove this unused "$page" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. MAJOR [MediaWikiService.php#L719](https://github.com/Wikia/app/blob/9132cb444f814e4ad657be02fed15fc60764d4d6/extensions/wikia/Search/classes/MediaWikiService.php#L719): Remove this unused "$file" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. MAJOR [WikiaException.php#L40](https://github.com/Wikia/app/blob/9132cb444f814e4ad657be02fed15fc60764d4d6/includes/wikia/nirvana/WikiaException.php#L40): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. MINOR [MediaWikiService.php#L169](https://github.com/Wikia/app/blob/9132cb444f814e4ad657be02fed15fc60764d4d6/extensions/wikia/Search/classes/MediaWikiService.php#L169): Define a constant instead of duplicating this literal "action" 6 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [MediaWikiService.php#L183](https://github.com/Wikia/app/blob/9132cb444f814e4ad657be02fed15fc60764d4d6/extensions/wikia/Search/classes/MediaWikiService.php#L183): Define a constant instead of duplicating this literal "query" 9 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [MediaWikiService.php#L586](https://github.com/Wikia/app/blob/9132cb444f814e4ad657be02fed15fc60764d4d6/extensions/wikia/Search/classes/MediaWikiService.php#L586): Define a constant instead of duplicating this literal "allmessages" 4 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [MediaWikiService.php#L679](https://github.com/Wikia/app/blob/9132cb444f814e4ad657be02fed15fc60764d4d6/extensions/wikia/Search/classes/MediaWikiService.php#L679): Define a constant instead of duplicating this literal "width" 9 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [MediaWikiService.php#L680](https://github.com/Wikia/app/blob/9132cb444f814e4ad657be02fed15fc60764d4d6/extensions/wikia/Search/classes/MediaWikiService.php#L680): Define a constant instead of duplicating this literal "height" 9 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

**Review by macbre — APPROVED** [review]

**Comment by rogatty** [comment]
Unit tests failed with:
```
1) SamplerProxyTest::testSampling50Percent
Failed asserting that 22 is equal to 20.0 or is less than 20.0.
```
but passed after a retry.
¯\\_(ツ)_/¯

**Pull request merged**